File: app/src/main/java/com/stevenfrew/beatprompter/cache/FileParseError.java
package com.stevenfrew.beatprompter.cache;

public class FileParseError
{
    private int mLineNumber=-1;
    private String mMessage;

    public FileParseError(Tag tag,String message)
    {
        this(tag==null?-1:tag.mLineNumber,message);
    }

    public FileParseError(int lineNumber,String message)
    {
       mLineNumber=lineNumber;
        mMessage=message;
    }

    public String getErrorMessage()
    {
        return (mLineNumber!=-1?""+mLineNumber+": ":"")+mMessage;
    }
}

File: app/src/main/kotlin/com/stevenfrew/beatprompter/cache/FileParseError.kt
package com.stevenfrew.beatprompter.cache

class FileParseError(lineNumber: Int, private val mMessage: String?) {
    private var mLineNumber = -1

    val errorMessage: String
        get() = (if (mLineNumber != -1) "$mLineNumber: " else "") + mMessage

    constructor(tag: Tag?, message: String?) : this(tag?.mLineNumber ?: -1, message)

    init {
        mLineNumber = lineNumber
    }
}